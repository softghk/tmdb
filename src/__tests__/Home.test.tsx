import React from 'react'
import {render, screen, fireEvent, act} from '@testing-library/react'
import Home from '../pages/index'
import {QueryClientWrapper} from '@layout'

describe('Home Page', () => {
  const HomeScreen = (
    <QueryClientWrapper>
      <Home />
    </QueryClientWrapper>
  )

  // Mock localStorage methods
  const localStorageMock = {
    getItem: jest.fn(),
    setItem: jest.fn(),
  }

  beforeAll(() => {
    Object.defineProperty(global, 'localStorage', {value: localStorageMock})
  })

  beforeEach(() => {
    // Clear mock function calls before each test
    localStorageMock.getItem.mockClear()
    localStorageMock.setItem.mockClear()
  })

  test('it should render the component', () => {
    const {container} = render(HomeScreen)
    expect(container).toMatchSnapshot()
  })
})
