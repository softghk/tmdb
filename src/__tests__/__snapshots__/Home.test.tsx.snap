// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page it should render the component 1`] = `
<div>
  <div
    class="max-w-[1170px] mx-auto p-10"
  >
    <h1
      class="text-center text-4xl mt-6 mb-16"
    >
      TMDB MOVIES
    </h1>
    <div
      class="text-white mb-10"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        data-testid="sort-select"
      >
        <div
          aria-controls=":r0:"
          aria-expanded="false"
          aria-haspopup="listbox"
          class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          role="combobox"
          tabindex="0"
        >
          DESC
        </div>
        <input
          aria-hidden="true"
          aria-invalid="false"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="desc"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="text-center my-10"
    >
      Loading ...
    </div>
  </div>
</div>
`;
